Mathlib/SetTheory/Ordinal/Arithmetic.lean

```diff
@@ -1331,6 +1314,63 @@ theorem bddAbove_iff_small {s : Set Ordinal.{u}} : BddAbove s ↔ Small.{u} s :=
   ⟨fun ⟨a, h⟩ => small_subset <| show s ⊆ Iic a from fun _ hx => h hx, fun _ =>
     bddAbove_of_small _⟩
 
+theorem bddAbove_range_comp {ι : Type u} {f : ι → Ordinal.{v}} (hf : BddAbove (range f))
+    (g : Ordinal.{v} → Ordinal.{max v w}) : BddAbove (range (g ∘ f)) := by
+  rw [range_comp, bddAbove_iff_small]
+  rw [bddAbove_iff_small] at hf
+  exact small_lift _
+
+theorem iSup_le_iff_of_small {ι : Type u} {f : ι → Ordinal.{v}} [Small.{v} ι] {a} :
+    iSup f ≤ a ↔ ∀ i, f i ≤ a :=
+  ciSup_le_iff' (Ordinal.bddAbove_iff_small.mpr (small_range f))
+
+theorem small_of_exists_injection {f : Ordinal.{u} → Ordinal.{v}} (h : f.Injective)
+    {S : Set Ordinal.{u}} [hs : Small.{u} S] : Small.{v} S := by
+  rcases hs.equiv_small with ⟨α, ⟨hα⟩⟩
+  rcases univLE_of_injective h α with ⟨β, ⟨hβ⟩⟩
+  rw [small_iff]
+  exact ⟨β, ⟨hα.trans hβ⟩⟩
+
+-- TODO: make sSup version.
+theorem IsNormal.map_iSup {f : Ordinal.{u} → Ordinal.{v}} (H : IsNormal f)
+    {ι : Type w} (g : ι → Ordinal.{u}) [Small.{u} ι] [Nonempty ι] :
+    f (⨆ i, g i) = ⨆ i, f (g i) := eq_of_forall_ge_iff fun a ↦ by
+  haveI : Small.{v} (range g) := small_of_exists_injection H.strictMono.injective
+  rw [H.le_set' Set.univ Set.univ_nonempty g]
+  · rw [← iSup_range', Ordinal.iSup_le_iff_of_small]
+    simp
+  · intro o
+    rw [Ordinal.iSup_le_iff_of_small]
+    simp
+
+theorem IsNormal.map_iSup_bddAbove {f : Ordinal.{u} → Ordinal.{v}} (H : IsNormal f)
+    {ι : Type w} (g : ι → Ordinal.{u}) (hg : BddAbove (range g))
+    [Nonempty ι] : f (⨆ i, g i) = ⨆ i, f (g i) := eq_of_forall_ge_iff fun a ↦ by
+  haveI : Small.{u} (range g) := bddAbove_iff_small.mp hg
+  haveI : Small.{v} (range g) := small_of_exists_injection H.strictMono.injective
```

Comment:
In Lean 4, `I` stands for "inline" rather than "instance". So a normal `have` block will work just fine here.
```suggestion
  have : Small.{u} (range g) := bddAbove_iff_small.mp hg
  have : Small.{v} (range g) := small_of_exists_injection H.strictMono.injective
```